x-pack/plugins/triggers_actions_ui/public/application/sections/rule_form/rule_form_consumer_selection.tsx

```diff
@@ -67,13 +67,20 @@ export interface RuleFormConsumerSelectionProps {
   consumers: RuleCreationValidConsumer[];
   onChange: (consumer: RuleCreationValidConsumer | null) => void;
   errors: IErrorObject;
-  selectedConsumer: RuleCreationValidConsumer | null | undefined;
+  selectedConsumer?: RuleCreationValidConsumer | null;
+  initialSelectedConsumer?: RuleCreationValidConsumer | null;
```

Comment:
Why is `null` also acceptable?

Reply:
that's the way it was setup before me so I kept it, I will ask @Zacqary if he has any use cases for it

Reply:
FYI -> @Zacqary

@maryam-saeidi had the idea to use `null` to avoid any kind of initialization. Therefore, rule creator can have the choice to obligated their user to set this value by themself.

Reply:
That works. We can differentiate between `null` as an explicit uninitialized empty value, and `undefined` as something waiting for its initial initialization.